src/components/VideoPlayer2.tsx

```diff
@@ -92,6 +92,52 @@ export const VideoPlayer: FunctionComponent<VideoPlayerProps> = ({
     return pipButtonContainer;
   };
 
+  // Fetch previously saved captions setting from localStorage
+  useEffect(() => {
```

Comment:
Why have 2 useEffects with same dependencies array? Can you add the below login in here only?

Reply:
yes sure